wasm/wasm-introduction.md

```diff
@@ -0,0 +1,157 @@
+interaction between the browser and the internet via the request-response life cycle. That's it, plain
+and simple. 
+## So, this scenario is working at the moment. What is the problem?
+The problems with the traditional web are the limitations of the browser as interpreter/compiler(Depending
+on the technology implemented by the browser) and renderer at the same time. In general, the performance of
+JavaScript running resource-intensive applications like video editors, 3D games, and so on, is far away to
+compete with the native performance. 
+Remember JavaScript was designed for humans and contains rules and redundancy that must be checked by the browser to run the source code. 
+## How to solve this problem? 
+One of the many solutions for this problem could be the inclusion of a new lower-level format with a compact
+binary format to run alongside JavaScript. In this way, we could use this language to develop near-native
+performance features.
+## WebAssembly, time to raise!
+Having these ideas in mind, [WebAssembly(a.k.a WASM)](https://webassembly.org/) was launched in 2017 as a [binary instruction format](https://webassembly.github.io/spec/core/binary/index.html)
+for a **stack-based virtual machine**. We know this definition is almost the same shown at [webassembly.org](https://webassembly.org/)
+so let extend a little bit more this explanation. Let's start with **binary instruction format**, this one is simple
+because it represents the way [WASM](https://webassembly.org/) exposes instructions to the computer to be processed. On
+the other hand, we have a **stack-based virtual machine** which stores the instructions from WASM files in memory
+using a stack as the main structure(We will talk more about this later in this post).
+Now that we are understanding a little bit more the concept of WebAssembly, so let's continue with the goals and benefits this technology brings to us.
+### Main goals of WebAssembly
+Like any other technology on the web, WebAssembly has been created thinking about some specific goals in mind. We are
+presenting these goals differently as follows:
+- Security: WebAssembly provides a secure sandboxed virtual machine. This means that we can WASM code from any web application
+without thinking about intrusions over our personal computers or servers. Proof of this point is that some of the most used browsers
+on the web have its support enabled by default because they are confident about the security provided by [WASM](https://webassembly.org/).
+- Fast execution: [WASM](https://webassembly.org/) is compiled into a binary format that will be executed at a low level. In this way, we will deliver
+near-native performance execution times. The format used by WASM is very compact so the browser can parse it as fast as possible.
+The translation of a [WebAsembly](https://webassembly.org/) module to a native code can be cached by the browser. In this way, the
```

Comment:
"... to native code...", no article required, code is generally uncountable in this context.